oracle/options.go

```diff
@@ -1,91 +1,98 @@
-	return func(o *OracleImpl) {
-		if maxCacheAge <= 0 {
-			panic("max cache age must be positive")
+// WithMarketMap sets the market map for the oracle.
+func WithMarketMap(marketMap mmtypes.MarketMap) Option {
+	return func(m *OracleImpl) {
+		if err := marketMap.ValidateBasic(); err != nil {
+			panic(err)
 		}
 
-		o.maxCacheAge = maxCacheAge
+		m.marketMap = marketMap
 	}
 }
 
-// WithLogger sets the logger on the Oracle.
-func WithLogger(logger *zap.Logger) Option {
-	return func(o *OracleImpl) {
-		if logger == nil {
-			panic("cannot set nil logger")
+// WithPriceAPIQueryHandlerFactory sets the Price API query handler factory for the oracle.
+// Specifically, this is what is utilized to construct price providers that are API based.
+func WithPriceAPIQueryHandlerFactory(factory types.PriceAPIQueryHandlerFactory) Option {
+	return func(m *OracleImpl) {
+		if factory == nil {
+			panic("api query handler factory cannot be nil")
 		}
 
-		o.logger = logger.With(zap.String("process", "oracle"))
+		m.priceAPIFactory = factory
 	}
 }
 
-// WithMetrics sets the metrics on the Oracle.
-func WithMetrics(metrics oraclemetrics.Metrics) Option {
-	return func(o *OracleImpl) {
-		if metrics == nil {
-			panic("cannot set nil metrics")
+// WithPriceWebSocketQueryHandlerFactory sets the websocket query handler factory for the oracle.
+// Specifically, this is what is utilized to construct price providers that are websocket based.
+func WithPriceWebSocketQueryHandlerFactory(factory types.PriceWebSocketQueryHandlerFactory) Option {
+	return func(m *OracleImpl) {
+		if factory == nil {
+			panic("websocket query handler factory cannot be nil")
 		}
 
-		o.metrics = metrics
+		m.priceWSFactory = factory
 	}
 }
 
-// WithMetricsConfig sets the metrics on the oracle from the given config.
-func WithMetricsConfig(config config.MetricsConfig) Option {
-	return func(o *OracleImpl) {
-		o.metrics = oraclemetrics.NewMetricsFromConfig(config)
+// WithMarketMapperFactory sets the market map factory for the oracle.
+// Specifically, this is what is utilized to construct market map providers.
+func WithMarketMapperFactory(factory mmclienttypes.MarketMapFactory) Option {
+	return func(m *OracleImpl) {
+		if factory == nil {
+			panic("market map factory cannot be nil")
+		}
+
+		m.marketMapperFactory = factory
 	}
 }
 
-// WithPriceAggregator sets the data aggregator on the Oracle.
-func WithPriceAggregator(agg PriceAggregator) Option {
-	return func(o *OracleImpl) {
-		if agg == nil {
-			panic("cannot set nil aggregator")
-		}
+// WithWriteTo sets the file path to which market map updates will be written to. Note that this is optional.
+func WithWriteTo(filePath string) Option {
+	return func(m *OracleImpl) {
+		m.writeTo = filePath
+	}
+}
 
-		o.priceAggregator = agg
+func WithPriceProviders(pps ...*types.PriceProvider) Option {
+	return func(m *OracleImpl) {
+		for _, pp := range pps {
+			m.priceProviders[pp.Name()] = ProviderState{Provider: pp}
+		}
 	}
 }
```

Comment:
Is this still necessary?

Reply:
so this was to allow mock providers to be passed into the oracle. i don't think we can stub in fake provider responses without this. see `providers_test.go`.